### PR TITLE
Add --version CLI option

### PR DESCRIPTION
**Pull request by RogerZhongAWS**
### Motivation
- Issue number: https://sim.amazon.com/issues/DC-3248

#### Change summary
Added CLI version output option.

#### Revision diff summary
Fixed some formatting to get the linter check to pass.

### Testing
Changes have been manually tested.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/config/Config.cpp [hunk 1]
```diff
@@ -1675,13 +1679,13 @@ void Config::PrintHelpMessage()
         "%s [true|false]:\t\t\t\t\tEnables/Disables Device Defender feature\n"
         "%s [true|false]:\t\t\t\tEnables/Disables Fleet Provisioning feature\n"
         "%s [true|false]:\t\t\t\t\t\tEnables/Disables Pub/Sub Sample feature\n"
-        "%s [true|false]:\t\t\t\t\t\tEnables/Disables Sample Shadow feature\n"
-        "%s [true|false]:\t\t\t\t\t\tEnables/Disables Config Shadow feature\n"
+        "%s [true|false]:\t\t\t\t\tEnables/Disables Sample Shadow feature\n"
```
Comment (HarshGandhi-AWS) [inline source/config/Config.cpp:1682]: thank you for updating this. It was pending for sometime now :D

**Review by HarshGandhi-AWS — APPROVED** [review]
Everything looks good to me.

File: source/config/Config.h [hunk 2]
```diff
@@ -333,6 +333,7 @@ namespace Aws
                 static constexpr char DEFAULT_SAMPLE_SHADOW_OUTPUT_DIR[] = "~/.aws-iot-device-client/sample-shadow/";
 
                 static constexpr char CLI_HELP[] = "--help";
+                static constexpr char CLI_VERSION[] = "--version";
```
Comment (marcoemorais-aws) [inline source/config/Config.h:336]: @RogerZhongAWS Can you add a short blurb to the Version doc in the repo that mentions customers can retrieve the version using the `--version` argument to the cli and include a short snippet.

https://github.com/awslabs/aws-iot-device-client/blob/main/docs/VERSION.md

Comment (RogerZhongAWS) [reply source/config/Config.h:336]: yes

**Review by marcoemorais-aws — CHANGES_REQUESTED** [review]
Change LGTM, merits some mention in the version doc.

**Review by marcoemorais-aws — APPROVED** [review]

**Review by shangabl — APPROVED** [review]

**Pull request merged**